### PR TITLE
Create 1911 Encyclopedia Britannica.js

### PR DESCRIPTION
**Pull request by guitarzxt**
A scraper for the 1911 Encyclopedia Britannica housed at jrank

### CODE REVIEW

**Comment by aurimasv** [comment]
Please use (Scaffold)[https://www.zotero.org/support/dev/translators/scaffold] to save the translator file and submit that file. Your file is missing a couple sections of the translator (e.g. see (Encyclopedia of Chicago)[https://github.com/zotero/translators/blob/master/Encyclopedia%20of%20Chicago.js])

File: 1911 Encyclopedia Britannica.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.-
+*/
+
+
+FW.Scraper({ 
+	itemType : 'encyclopediaArticle',
+	encyclopediaTitle : "1911 Encyclopædia Britannica",
+	detect : FW.Url().match(/(encyclopedia\.jrank\.org\/).{7}(\/).+/),
```
Comment (aurimasv) [inline 1911 Encyclopedia Britannica.js]: You don't need the capturing parentheses in your regexp

File: 1911 Encyclopedia Britannica.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	title : FW.Xpath('//html/body/div[1]/div[3]/h1').text(),
```
Comment (aurimasv) [inline 1911 Encyclopedia Britannica.js]: probably better to select by `@class=articleName` and ditch the full path to the element. Would be less fragile.

File: 1911 Encyclopedia Britannica.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	attachments : [{
+		url : FW.Url(),
+		title : "Jrank Snapshot",
+		type : "text/html"
+		}],
```
Comment (aurimasv) [inline 1911 Encyclopedia Britannica.js]: de-indent this line.

File: 1911 Encyclopedia Britannica.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	abstractNote : FW.Xpath('//html/body/div[1]/blockquote[1]/b').text().trim(),
```
Comment (aurimasv) [inline 1911 Encyclopedia Britannica.js]: This xpath doesn't seem to work very well, e.g. http://encyclopedia.jrank.org/CAL_CAR/CALENDAR.html

File: 1911 Encyclopedia Britannica.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+	publisher : "Encyclopædia Britannica, Inc.",
+	place : "New York",
+	edition : "11th ed.",
+	numberOfVolumes : '29',
+	volume : FW.Xpath('//html/body/div[1]/div[3]/div[@class="tinyText"]').text().remove(/(Originally appearing in Volume\nV)/).remove(/(,)/).remove(/\n/).remove(/(P).+/).remove(/\n/).remove(/(o).+/).prepend("Vol. "),
+	hooks : { "scraperDone": function (item,doc, url) {
+		item.creators.push({
+		"firstName" : "Hugh",
+		"lastName" : "Chisholm",
+		"creatorType" : "editor"
```
Comment (aurimasv) [inline 1911 Encyclopedia Britannica.js]: indent block

File: 1911 Encyclopedia Britannica.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+		})
+		var volNum = (item.volume).substring(5,7);
```
Comment (aurimasv) [inline 1911 Encyclopedia Britannica.js]: Why aren't we constructing this correctly in the first place? Looks like you're removing the "Vol. " that is specifically prepended on L40.

nit: don't need the parentheses

File: 1911 Encyclopedia Britannica.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+		if (volNum <= 9){
```
Comment (aurimasv) [inline 1911 Encyclopedia Britannica.js]: nit: space before `{` and after `}` below

File: 1911 Encyclopedia Britannica.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+			item.date = 1910;
+		}else {item.date = 1911}
```
Comment (aurimasv) [inline 1911 Encyclopedia Britannica.js]: if you're using  `{ ... }` for some of the `if` statement, use it for all of it

File: 1911 Encyclopedia Britannica.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+	
+		if (item.abstractNote == "There are no comments yet for this article."){
+			item.abstractNote = "";
+		}
+	}} 
+});
```
Comment (aurimasv) [inline 1911 Encyclopedia Britannica.js]: Please add tests (this will be easy when using Scaffold)

File: 1911 Encyclopedia Britannica.js [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+		type : "text/html"
+		}],
+	abstractNote : FW.Xpath('//html/body/div[1]/blockquote[1]/b').text().trim(),
+	archive : "Jrank",
+	rights : FW.Xpath('//div[@id="footer"]').text(),
+	publisher : "Encyclopædia Britannica, Inc.",
+	place : "New York",
+	edition : "11th ed.",
+	numberOfVolumes : '29',
+	volume : FW.Xpath('//html/body/div[1]/div[3]/div[@class="tinyText"]').text().remove(/(Originally appearing in Volume\nV)/).remove(/(,)/).remove(/\n/).remove(/(P).+/).remove(/\n/).remove(/(o).+/).prepend("Vol. "),
```
Comment (aurimasv) [inline 1911 Encyclopedia Britannica.js]: please split into multiple lines

Comment (aurimasv) [inline 1911 Encyclopedia Britannica.js]: Do not prepend the "Vol. " part. item.volume should just be the number.

File: 1911 Encyclopedia Britannica.js [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+		url : FW.Url(),
+		title : "Jrank Snapshot",
+		type : "text/html"
+		}],
+	abstractNote : FW.Xpath('//html/body/div[1]/blockquote[1]/b').text().trim(),
+	archive : "Jrank",
+	rights : FW.Xpath('//div[@id="footer"]').text(),
+	publisher : "Encyclopædia Britannica, Inc.",
+	place : "New York",
+	edition : "11th ed.",
```
Comment (aurimasv) [inline 1911 Encyclopedia Britannica.js]: Just "11"

File: 1911 Encyclopedia Britannica.js [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+	attachments : [{
+		url : FW.Url(),
+		title : "Jrank Snapshot",
+		type : "text/html"
+		}],
+	abstractNote : FW.Xpath('//html/body/div[1]/blockquote[1]/b').text().trim(),
+	archive : "Jrank",
+	rights : FW.Xpath('//div[@id="footer"]').text(),
+	publisher : "Encyclopædia Britannica, Inc.",
+	place : "New York",
```
Comment (aurimasv) [inline 1911 Encyclopedia Britannica.js]: "New York, New York, USA" (or something to that extent)

**Comment by aurimasv** [comment]
Maybe a more appropriate translator title here would be "Jrank Online Encyclopedia", since according to http://encyclopedia.jrank.org/ it includes several encyclopedias.

File: 1911 Encyclopedia Britannica.js (hunk 4 above)
Comment (aurimasv) [inline 1911 Encyclopedia Britannica.js]: OK, I see what you're scraping here and that you're covering this case later. But based on http://encyclopedia.jrank.org/CAL_CAR/CALHOUN_JOHN_CALDWELL_1782_185o.html this is not the sort of information that should be put into abstractNote. You can put it in a note attachment though, if you think this is useful information to maintain.

Comment (guitarzxt) [inline 1911 Encyclopedia Britannica.js]: What is a note attachment?

File: 1911 Encyclopedia Britannica.js (hunk 10 above)
Comment (guitarzxt) [inline 1911 Encyclopedia Britannica.js]: Does Zotero add Vol. in front when the citation is added into the paper?

File: 1911 Encyclopedia Britannica.js (hunk 11 above)
Comment (guitarzxt) [inline 1911 Encyclopedia Britannica.js]: Is the ed. added by Zotero into the citation?

**Comment by AbeJellinek** [comment]
This:

1. Hasn't seen activity since 2016.
2. Is a translator for a specific item on a larger site.
3. Uses FW.
4. Doesn't have tests.

So I'm going to close for now. I'm very happy to reopen if it's possible to fix these issues.

**Pull request closed without merging**